condacolab.py

```diff
@@ -82,19 +108,31 @@ def install_from_url(
         shutil.copyfileobj(response, out)
 
     print("📦 Installing...")
-    task = run(
+
+    condacolab_task = run_subprocess(
         ["bash", installer_fn, "-bfp", str(prefix)],
-        check=False,
-        stdout=PIPE,
-        stderr=STDOUT,
-        text=True,
-    )
-    os.unlink(installer_fn)
-    with open("condacolab_install.log", "w") as f:
-        f.write(task.stdout)
-    assert (
-        task.returncode == 0
-    ), "💥💔💥 The installation failed! Logs are available at `/content/condacolab_install.log`."
+        "condacolab_install.log",
+        )
+
+    """
+    Installing the following packages because Colab server expects these packages to be installed in order to launch a Python kernel :
+        - matplotlib-base
+        - psutil
+        - google-colab
+        - colabtools
+    """
+
+    conda_exe = "mamba" if os.path.isfile(f"{prefix}/bin/mamba") else "conda"
+
+    conda_task = run_subprocess(
```

Comment:
Maybe one way we can speed up certain things is by querying the environment with `conda list -p {prefix} --json` and then exploring the JSON dictionary to see if we can find these required packages. 

It would be something like:

```python
output = check_output([f"{prefix}/bin/conda", "list", "--json"])
payload = json.loads(output)
installed_names = {pkg["name"] for pkg in payload} # not really sure this is the payload schema, but you get the idea
required_packages = ["matplotlib-base", "psutil", "google-colab"]
for pkg in required_packages.copy():
  if pkg in installed_names:
    required_packages.remove(pkg)

# then the command to conda install would be
if required_packages:  # only if the list hasn't been emptied
  # conda install -yq *required_packages

```